src/main/java/seedu/hirehub/logic/commands/AddApplicationCommand.java

```diff
@@ -0,0 +1,155 @@
+            + "and application has been processed already into the list";
+    public static final String MESSAGE_NO_PERSON_IN_LIST = "This candidate is not in the current list of candidates! ";
+    public static final String MESSAGE_NO_JOB_IN_LIST = "This job is not yet created! ";
+
+    private final Email email;
+    private final String jobTitle;
+    private final Status status;
+
+    /**
+     * @param email of the person in the filtered person list to add a job application for
+     * @param jobTitle of the job to match a candidate with the job they applied for
+     * @param status of the person to show the current recruitment status of a candidate
+     */
+    public AddApplicationCommand(Email email, String jobTitle, Status status) {
+        requireNonNull(email);
+        requireNonNull(jobTitle);
+        requireNonNull(status);
+        this.email = email;
+        this.jobTitle = jobTitle;
+        this.status = status;
+    }
+
+    /**
+     * Overloads the constructor where the status is set as "PRESCREEN" as a default if it is not
+     * provided by the recruiter
+     */
+    public AddApplicationCommand(Email email, String jobTitle) {
+        requireNonNull(email);
+        requireNonNull(jobTitle);
+        this.email = email;
+        this.jobTitle = jobTitle;
+        this.status = new Status("PRESCREEN");
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        Person personMatchingEmail = findPersonMatchingEmail(email, model);
+        Job jobMatchingTitle = findJobMatchingTitle(jobTitle, model);
+
+        Application newCandidateApplication = new Application(personMatchingEmail, jobMatchingTitle, status);
+
+        if (model.hasApplication(newCandidateApplication)) {
+            throw new CommandException(MESSAGE_DUPLICATE_APPLICATION);
+        }
+
+        model.addApplication(newCandidateApplication);
+
+        // TODO: remove print statements once UI is up and running
+        for (Application app: model.getFilteredApplicationList()) {
+            System.out.println(app);
+        }
+
+        return new CommandResult(String.format(MESSAGE_ADD_SUCCESS, personMatchingEmail.getName(), jobTitle));
+    }
+
+    /**
+     * Finds a person with the given email in the filtered person list
+     * @param email for email of a candidate to search for in the list
+     * @param model to retrieve the filtered person list
+     */
+    public Person findPersonMatchingEmail(Email email, Model model) throws CommandException {
+        requireNonNull(email);
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        for (int i = 0; i < lastShownList.size(); i++) {
+            if (email.equals(lastShownList.get(i).getEmail())) {
+                return lastShownList.get(i);
+            }
```

Comment:
I noticed you compared emails in the findPersonMatchingEmail method but the isSameJob method is invoked in findJobMatchingTitle. Can I check if this is intended?

Reply:
Yes

Reply:
noted